src/cli.rs

```diff
@@ -296,6 +303,11 @@ impl Cli {
         match Self::from_iter_safe(iter) {
             Ok(cli) => cli,
             Err(err) if err.kind == ErrorKind::HelpDisplayed => {
+                let mut app = if env::var("NO_COLOR").is_ok() {
```

Comment:
After experimenting a bit I think this is the best way out. It's possible to [look at](https://docs.rs/termcolor/1.1.2/termcolor/trait.WriteColor.html#tymethod.supports_color) `termcolor`'s detection, but `NO_COLOR` is really the only piece that's missing.

One nitpick, so unicode won't matter:
```suggestion
                let mut app = if env::var_os("NO_COLOR").is_some() {
```